File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.py
import math

def solution(n, k):
    answer = []
    numbers = [i for i in range(1, n + 1)]
    
    k -= 1
    
    while n > 0:
        total_case = math.factorial(n - 1)
        
        index = k // total_case
        
        answer.append(numbers.pop(index))
        
        k %= total_case
        n -= 1
        
    return answer

File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.js
function solution(n, k) {
    const answer = [];
    const numbers = Array.from({ length: n }, (_, i) => i + 1);
    
    let factorial = 1;
    for (let i = 1; i < n; i++) factorial *= i;

    k--;

    while (numbers.length > 0) {
        const index = Math.floor(k / factorial);
        answer.push(numbers[index]);
        
        numbers.splice(index, 1);

        if (numbers.length > 0) {
            k %= factorial;
            factorial /= numbers.length;
        }
    }

    return answer;
}